### PR TITLE
feat: API added to check if the release is installed or not with some refactoring

### PR DESCRIPTION
**Pull request by manish-agrawal-ai**
# Description
API added to check if the release is installed or not with some refactoring

Fixes # https://github.com/devtron-labs/devtron/issues/1270

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
New API hit

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: pkg/service/HelmAppService.go [hunk 1]
```diff
@@ -534,6 +500,8 @@ func (impl HelmAppServiceImpl) InstallRelease(ctx context.Context, request *clie
 		UpgradeCRDs:      true,
 		CreateNamespace:  true,
 	}
+
+	impl.logger.Debug("Installing release")
```
Comment (nishant-d) [inline pkg/service/HelmAppService.go:504]: add some contextual info in log

File: pkg/util/HelmServiceUtil.go [hunk 2]
```diff
@@ -2,10 +2,57 @@ package util
...
+// app health is worst of the nodes health
+func BuildAppHealthStatus(nodes []*bean.ResourceNode) *bean.HealthStatusCode {
+	appHealthStatus := bean.HealthStatusHealthy
+
+	for _, node := range nodes {
+		nodeHealth := node.Health
+		if nodeHealth == nil {
+			continue
+		}
+		if health.IsWorse(health.HealthStatusCode(appHealthStatus), health.HealthStatusCode(nodeHealth.Status)) {
```
Comment (nishant-d) [inline pkg/util/HelmServiceUtil.go:52]: are we using gitops engine as dependency ?

File: grpc/applist.proto [hunk 3]
```diff
@@ -29,6 +29,7 @@ service ApplicationService {
   rpc GetDeploymentDetail(DeploymentDetailRequest) returns (DeploymentDetailResponse){}
   rpc InstallRelease(InstallReleaseRequest) returns (InstallReleaseResponse){}
   rpc UpgradeReleaseWithChartInfo(InstallReleaseRequest) returns (UpgradeReleaseResponse){}
+  rpc IsReleaseInstalled(ReleaseIdentifier) returns (BooleanResponse){}
```
Comment (nishant-d) [inline grpc/applist.proto:32]: can we name it better ?

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: pkg/service/HelmAppService.go (hunk 1 above)
Comment (manish-agrawal-ai) [reply pkg/service/HelmAppService.go:504]: Contextual info is logged in the caller method in HelmAppServiceWrapper.go class. This log is just for to know which step currently code is executing (for debugging)

File: pkg/util/HelmServiceUtil.go (hunk 2 above)
Comment (manish-agrawal-ai) [reply pkg/util/HelmServiceUtil.go:52]: yes. (This is just refactoring. moved from here to there with no changes. )

File: grpc/applist.proto (hunk 3 above)
Comment (manish-agrawal-ai) [reply grpc/applist.proto:32]: Any suggestion? I think this name is enough self explanatory.

File: pkg/service/HelmAppService.go (hunk 1 above)
Comment (nishant-d) [reply pkg/service/HelmAppService.go:504]: having contextual info in log will help in debugging

Comment (manish-agrawal-ai) [reply pkg/service/HelmAppService.go]: done

**Review by nishant-d — APPROVED** [review]

**Pull request merged**